### ISSUE TITLE
Problems with parsing datetime on serverside with generated serverside and clientside Java code

### ISSUE THREAD

**Issue by Szdnez** [swagger-api/swagger-codegen#3648]
##### Description

I have a swagger definition like below and generated serverside code and generated client side code with the swagger editor, run the server code and try to call the get-method, it always results in a failure that the called method could not be found.
When you debug on serverside you can see, that the problem is the parsing of the datetime parameter, a format exception is thrown while _new java.util.Date( String )_.

##### Swagger-codegen version

Swagger Editor Version 2.10.3

##### Swagger declaration file content or url

```YAML
/services:
    get:
      parameters:
        - name: begin
          in: query
          required: true
          type: string
          format: date-time
      responses:
        200:
          schema:
            type: object
            properties:
              response_code:
                $ref: '#/definitions/Responsecode'
              services:
                type: array
                items:
                  $ref: '#/definitions/Service'
```

##### Command line used for generation

- Swagger editor (Generate Server -> JAX-RS) 
- Swagger editor (Generate Client -> Java)

##### Steps to reproduce

- Swagger editor (Generate Server -> JAX-RS) 
- Swagger editor (Generate Client -> Java)

##### Related issues

There is also a problem with the date format (property with type string and format date). When you send such data as an answer from the server to the client, you get en error from the yoda framework, that the given "string" is not parseable for _DateLocal_. When you replace the _DateLocal_ via _java.util.Date_ in the clientside API everything works fine.

##### Suggest a Fix

To solve the problem I tried to change the generated clientside code. In the API-class I used _java.util.Date_ instead of _DateTime_ (yoda-framework). But it does not work, too.
Afterwards I tried to change the generated serverside code. In the API-class I changed the parametertype of the parameter _begin_ from _java.util.Date_ to _DateTime_ (yoda-Framework). But after that I get a new error during starting the server: _Missing dependency for method public javax.ws.rs.core.Response gen.java.io.swagger.api.ServicesApi.servicesGet(java.lang.String,org.joda.time.DateTime) throws gen.java.io.swagger.api.NotFoundException at parameter at index 1_.

Because of the related issue I think it is a problem in the serverside generated code.

**Comment by cbornet** [comment]
Can you paste your request payload ?

**Comment by Szdnez** [comment]
> Can you paste your request payload ?
Where can I see it or where I hav to set a break point to copy it? I can not see it in console per default or something like that.

**Comment by cbornet** [comment]
I didn't understand that you were using the codegen for both client and server.
You can forget my question.

**Comment by Szdnez** [comment]
Besides of this issue I also found another bug. Outgoing strings of timestamps from server to client (on server-side _java.util.Date_) does not fulfill the RFC requirements. There should be a : between hours and minutes in the offset. But it is missing. For example you get 2016-09-01T06:00:00.000+0000 instead of 2016-09-01T06:00:00.000+00**:**00.

**Comment by cbornet** [comment]
Yes. That's because the JAX-RS template uses the mapper from io.swagger.uti.Json which doesn't change the default Date format. And the default Date format is "yyyy-MM-dd'T'HH:mm:ss.SSSZ" which is not rfc1339 compatible (should be ""yyyy-MM-dd'T'HH:mm:ss.SSSXXX"). That's indeed a bug.
I guess that's the same issue for client to JAX-RS requests.

**Comment by cbornet** [comment]
And also the JAX-RS codegen doesn't support date (without time) type if you use java.util.Date as dateLibrary.
I advise to generate the JAX-RS server code with --dateLibrary=joda (or even better : java8 if you can).
@wing328 shouldn't we make joda the default date lib for JAX-RS ?

**Comment by Szdnez** [comment]
@cbornet I think so too. The problem is the usage of _java.util.Date_, where the string-handling is not compatibel to RFC-format.
Til now I only use the editor from _swagger.io_ and its generation. There is no possibility to set any parameter.
